modules/porous_flow/src/dirackernels/PorousFlowPointEnthalpySourceFromPostprocessor.C

```diff
@@ -0,0 +1,56 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PorousFlowPointEnthalpySourceFromPostprocessor.h"
+#include "SinglePhaseFluidProperties.h"
+
+registerMooseObject("PorousFlowApp", PorousFlowPointEnthalpySourceFromPostprocessor);
+
+InputParameters
+PorousFlowPointEnthalpySourceFromPostprocessor::validParams()
+{
+  InputParameters params = DiracKernel::validParams();
+  params.addRequiredParam<PostprocessorName>("mass_flux",
+                                             "The postprocessor name holding the mass flux at this "
+                                             "point in kg/s (only positive values are allowed)");
```

Comment:
"only positive values are allowed".  You don't check for positivity.  Do you want to?  Or do you want to change to "please ensure this is positive so that this object acts like a source"?

Reply:
I don't want to check positivity at the moment. It happens to us that for very short time the mass flux is actually negative. Eventually, I would like to add logic that would look at the direction and did the right thing.

So I will change the doco.